packages/integration-tests/src/test-project-setup/conversation_handler_project.ts

```diff
@@ -320,6 +327,71 @@ class ConversationHandlerTestProject extends TestProjectBase {
     );
   };
 
+  private assertDefaultConversationHandlerCanExecuteTurnWithClientTool = async (
+    backendId: BackendIdentifier,
+    accessToken: string,
+    graphqlApiEndpoint: string,
+    apolloClient: ApolloClient<NormalizedCacheObject>
+  ): Promise<void> => {
+    const defaultConversationHandlerFunction = (
+      await this.resourceFinder.findByBackendIdentifier(
+        backendId,
+        'AWS::Lambda::Function',
+        (name) => name.includes('default')
+      )
+    )[0];
+
+    // send event
+    const event: ConversationTurnEvent = {
+      conversationId: randomUUID().toString(),
+      currentMessageId: randomUUID().toString(),
+      graphqlApiEndpoint: graphqlApiEndpoint,
+      messages: [
+        {
+          role: 'user',
+          content: [
+            {
+              text: 'What is the temperature in Seattle?',
+            },
+          ],
+        },
+      ],
+      request: {
+        headers: { authorization: accessToken },
+      },
+      toolsConfiguration: {
+        clientTools: [
+          {
+            name: 'thermometer',
+            description: 'Provides the current temperature for a given city.',
+            inputSchema: {
+              json: {
+                type: 'object',
+                properties: {
+                  city: {
+                    type: 'string',
+                    description: 'string',
+                  },
+                },
+                required: [],
+              },
+            },
+          },
+        ],
+      },
+      ...commonEventProperties,
+    };
+    const response = await this.executeConversationTurn(
+      event,
+      defaultConversationHandlerFunction,
+      apolloClient
+    );
+    // Assert that tool was used. I.e. that LLM used value returned by the tool.
+    assert.match(response.content, /Seattle/);
```

Comment:
Am I understanding correctly that the event messages are included in the response? If not, how are we sure the response will always include `Seattle` instead of something like `The temperature is x`?